HealthMonitoring/HealthMonitoring.csproj

```diff
@@ -46,6 +54,10 @@
     <Compile Include="Model\EndpointStats.cs" />
     <Compile Include="Model\EndpointStatus.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="PluginDiscovery.cs" />

```

Comment:
I am not sure if it belongs here. All other classes are either interfaces or POCO models, while PluginDiscovery is a specific implementation class bringing dependencies on filesystem and logging.
I would think rather about introducing a separate assembly for a common code that could be shared between Hosts, maybe something like HealthMonitoring.Hosting ?